datalad/core/local/create.py

```diff
@@ -365,6 +377,10 @@ def __call__(
         if initopts is not None and isinstance(initopts, list):
             initopts = {'_from_cmdline_': initopts}
 
+        # Inherit annex.private=true if parent dataset is given explicitly
+        if ds is not None:
+            private = private or ds.config.get('annex.private', None) == 'true'
```

Comment:
Minor hint: You may want to use `ConfigManager`'s `getbool` here, instead of string comparison. That would allow for `True`, `true`, `1`, etc. - see `anything2bool`.